exchanges/gateio/gateio_wrapper.go

```diff
@@ -2013,60 +1814,59 @@ func (g *Gateio) checkInstrumentAvailabilityInSpot(instrument currency.Pair) (bo
+	if !g.SupportsAsset(a) {
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, a)
+	}
+	settle, err := getSettlementCurrency(a, currency.EMPTYPAIR)
+	if err != nil {
+		return nil, err
 	}
-	switch item {
-	case asset.Futures:
-		var resp []futures.Contract
-		for k := range settlementCurrencies {
-			contracts, err := g.GetAllFutureContracts(ctx, settlementCurrencies[k])
+	switch a {
+	case asset.CoinMarginedFutures, asset.USDTMarginedFutures:
+		contracts, err := g.GetAllFutureContracts(ctx, settle)
+		if err != nil {
+			return nil, err
+		}
+		resp := make([]futures.Contract, len(contracts))
+		for i := range contracts {
+			name, err := currency.NewPairFromString(contracts[i].Name)
 			if err != nil {
 				return nil, err
 			}
-			contractsToAdd := make([]futures.Contract, len(contracts))
-			for j := range contracts {
-				var name currency.Pair
-				name, err = currency.NewPairFromString(contracts[j].Name)
-				if err != nil {
-					return nil, err
-				}
-				contractSettlementType := futures.Linear
-				switch {
-				case name.Base.Equal(currency.BTC) && settlementCurrencies[k].Equal(currency.BTC):
-					contractSettlementType = futures.Inverse
-				case !name.Base.Equal(settlementCurrencies[k]) && !settlementCurrencies[k].Equal(currency.USDT):
-					contractSettlementType = futures.Quanto
-				}
-				c := futures.Contract{
-					Exchange:             g.Name,
-					Name:                 name,
-					Underlying:           name,
-					Asset:                item,
-					IsActive:             !contracts[j].InDelisting,
-					Type:                 futures.Perpetual,
-					SettlementType:       contractSettlementType,
-					SettlementCurrencies: currency.Currencies{settlementCurrencies[k]},
-					Multiplier:           contracts[j].QuantoMultiplier.Float64(),
-					MaxLeverage:          contracts[j].LeverageMax.Float64(),
-				}
-				if contracts[j].FundingRate > 0 {
-					c.LatestRate = fundingrate.Rate{
-						Time: contracts[j].FundingNextApply.Time().Add(-time.Duration(contracts[j].FundingInterval) * time.Second),
-						Rate: contracts[j].FundingRate.Decimal(),
-					}
+			contractSettlementType := futures.Linear
+			switch {
+			case name.Base.Equal(currency.BTC) && settle.Equal(currency.BTC):
+				contractSettlementType = futures.Inverse
+			case !name.Base.Equal(settle) && !settle.Equal(currency.USDT):
+				contractSettlementType = futures.Quanto
+			}
+			c := futures.Contract{
+				Exchange:             g.Name,
+				Name:                 name,
+				Underlying:           name,
+				Asset:                a,
+				IsActive:             !contracts[i].InDelisting,
+				Type:                 futures.Perpetual,
+				SettlementType:       contractSettlementType,
+				SettlementCurrencies: currency.Currencies{settle},
+				Multiplier:           contracts[i].QuantoMultiplier.Float64(),
+				MaxLeverage:          contracts[i].LeverageMax.Float64(),
+			}
+			if contracts[i].FundingRate > 0 {
```

Comment:
Let's remove this `if` and keep the tasty innards. Negative funding rates exist, and potentially 0 - I haven't checked for this exchange, but its bad code I wrote

Reply:
Fixed https://github.com/gbjk/gocryptotrader/commit/8293184ee6